protocols/v2/noise-sv2/src/aed_cipher.rs

```diff
@@ -1,16 +1,60 @@
+// # AEAD Cipher
+//
+// Defines the [`AeadCipher`] trait which standardizes the interface for Authenticated Encryption
+// with Associated Data (AEAD) ciphers used within the Noise protocol implementation. The trait is
+// implemented for two common ciphers, [`ChaCha20Poly1305`] and [`Aes256Gcm`], providing encryption
+// and decryption functionality with authenticated data.
+//
+// ## Overview
+//
+// AEAD ciphers provide both confidentiality and integrity by encrypting data and creating an
+// authentication tag in a single step. The integrity of additional associated data (AAD) can also
+// be verified without including it in the encrypted output.
+//
+// ## Usage
+//
+// The [`AeadCipher`] trait is used by the [`crate::handshake::HandshakeOp`] trait to perform
+// cryptographic operations during the Noise protocol handshake, ensuring secure communication
+// between parties.
+
 use aes_gcm::Aes256Gcm;
 use chacha20poly1305::{aead::Buffer, AeadInPlace, ChaCha20Poly1305, ChaChaPoly1305, KeyInit};
 
+// Defines the interface for AEAD ciphers.
+//
+// The [`AeadCipher`] trait provides a standard interface for initializing AEAD ciphers, and for
+// performing encryption and decryption operations with additional Authenticated Associated Data (AAD). This trait is implemented
+// by either the [`ChaCha20Poly1305`] or [`Aes256Gcm`] specific cipher types, allowing them to be
+// used interchangeably in cryptographic protocols. It is utilized by the
+// [`crate::handshake::HandshakeOp`] trait to secure the handshake process.
+//
+// The `T: Buffer` represents the data buffer to be encrypted or decrypted. The buffer must
+// implement the [`Buffer`] trait, which provides necessary operations for in-place encryption and
+// decryption.
```

Comment:
Fixing the character length:

```suggestion
// Defines the interface for AEAD ciphers.
//
// The [`AeadCipher`] trait provides a standard interface for initializing AEAD ciphers, and for
// performing encryption and decryption operations with additional AAD. This trait is implemented
// by either the [`ChaCha20Poly1305`] or [`Aes256Gcm`] specific cipher types, allowing them to be
// used interchangeably in cryptographic protocols. It is utilized by the
// [`crate::handshake::HandshakeOp`] trait to secure the handshake process.
//
// The `T: Buffer` represents the data buffer to be encrypted or decrypted. The buffer must
// implement the [`Buffer`] trait, which provides necessary operations for in-place encryption and
// decryption.
```

Reply:
Somebody in previous reviews suggested to elaborate AAD.

Reply:
Yea I think it is best to keep it `Authenticated Associated Data (AAD)`